dds/DCPS/XTypes/TypeObject.h

```diff
@@ -1959,25 +1988,35 @@ bool operator>>(Serializer& strm, XTypes::Sequence<T>& seq)
       return false;
     }
   }
-  return true;
+  return (total_size == 0) ? true : strm.skip(end_of_seq - strm.pos());
```

Comment:
I'm not totally sure if this function handles the cases when the received data has a DHEADER but we skip the DHEADER, and when the received data doesn't have a DHEADER but we still read DHEADER. The two other cases (received data has DHEADER & we read DHEADER, and received data doesn't have DHEADER & we skip it) are handled correctly though.

Reply:
It turns out that deserializing with this "skip sequence DHEADER" mode wasn't needed.  I'll take it out and simplify this function.